modules/grafana/manifests/dashboards/deployment_dashboard.pp

```diff
@@ -23,6 +23,24 @@
   $app_domain = undef,
   $has_workers = true,
 ) {
+  if $has_workers {
+    $worker_row = [["worker_failures", "worker_successes"]]
+  } else {
+    $worker_row = []
+  }
+
+  $panel_partials = concat(
+    [
+      ["processor_count", "5xx_rate"],
+      ["error_deploy_counts", "recent_500_count", "period_500_count", "links"]
+    ],
+    $worker_row,
+    [
+      ["errors_by_controller_action"],
+      ["response_times_by_controller"]
+    ]
+  )
```

Comment:
This feel like a complicated way of building this list with the worker logic out the front. Any reason you didn't go with something simpler like:

```
$panel_partials = []
$panel_partials << ["processor_count", "5xx_rate"]
$panel_partials << ["error_deploy_counts", "recent_500_count", "period_500_count", "links"]
$panel_partials << ["worker_failures", "worker_successes"] if $has_workers
$panel_partials << ["errors_by_controller_action"]
$panel_partials << ["response_times_by_controller"]
```